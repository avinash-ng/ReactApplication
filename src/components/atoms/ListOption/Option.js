import React from 'react'
import PropTypes from 'prop-types'
const Option = props => {
    return (
        <option value={props.value}>{props.name}</option>
    )    
}

Option.propTypes = {
    value: PropTypes.string.isRequired,
    name:  PropTypes.string.isRequired
}
export default Option;
