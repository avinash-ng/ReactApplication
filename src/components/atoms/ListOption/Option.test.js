import Option from './Option'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Option /> ', () => {
    it('renders Option element', () => {
        const name = "delete"
        const value = "delete"
        const RenderComponent = shallow(
            <Option value={value} name={name}></Option>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly', () => {
        const name = "upload"
        const value="upload"
        
        const RenderComponent = shallow(<Option value={value} name={name}></Option>)
        expect(RenderComponent.find('option').text()).toBe(name)

    })


    it('Matches snapshot', () => {
        const value = "edit"
        const name = "edit"
        const RenderComponent = shallow(<Option value={value} name={name}></Option>)
        expect(RenderComponent).toMatchSnapshot()
    
    })
})