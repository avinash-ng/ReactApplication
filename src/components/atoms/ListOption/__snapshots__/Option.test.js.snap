// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option />  Matches snapshot 1`] = `ShallowWrapper {}`;
